src/Playwright.Tests.TestServer/SimpleServer.cs

```diff
@@ -107,7 +109,10 @@ public SimpleServer(int port, string contentRoot, bool isHttps)
                             if (context.WebSockets.IsWebSocketRequest)
                             {
                                 var webSocket = await context.WebSockets.AcceptWebSocketAsync();
-                                await webSocket.SendAsync(new(Encoding.UTF8.GetBytes("incoming")), WebSocketMessageType.Text, true, CancellationToken.None);
+                                if (_onWebSocketConnectionData != null)
```

Comment:
this adds the same functionality like our upstream webserver has to the .NET webserver